src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs

```diff
@@ -1174,6 +1215,15 @@ void ThrowIfNotPeekLockMode()
             }
         }
 
+        void ThrowIfSessionLockLost()
+        {
+            if (this.LinkError != null &&
+                this.LinkError.Condition.Equals(AmqpClientConstants.SessionLockLostError))
+            {
```

Comment:
I was thinking about that too, but at this point I'm not sure what the other something else is. So until we see issues and fully understand why we need to handle that scenario, I wanted to keep it defined to the scenario we know of.